packages/xo-server-backup-reports/src/index.js

```diff
@@ -100,36 +102,135 @@ const isSkippedError = error =>
   if (end !== undefined) {
-    markdown.push(`${indent}- **End time**: ${formatDate(end)}`)
+    markdown.push(`- **End time**: ${formatDate(end)}`)
     const duration = end - start
     if (duration >= 1) {
-      markdown.push(`${indent}- **Duration**: ${formatDuration(duration)}`)
+      markdown.push(`- **Duration**: ${formatDuration(duration)}`)
     }
   }
   return markdown
 }
 
-const addWarnings = (text, warnings, nbIndent = 0) => {
-  if (warnings === undefined) {
+const getWarningsMarkdown = (warnings = []) =>
+  warnings.map(({ message }) => `- **${ICON_WARNING} ${message}**`)
+
+const getErrorMarkdown = task => {
+  let message
+  if (
+    task.status === 'success' ||
+    (message = task.result?.message ?? task.result?.code) === undefined
+  ) {
     return
   }
 
-  const indent = INDENT.repeat(nbIndent)
-  warnings.forEach(({ message }) => {
-    text.push(`${indent}- **${ICON_WARNING} ${message}**`)
-  })
+  const label = task.status === 'skipped' ? 'Reason' : 'Error'
+  return `- **${label}**: ${message}`
+}
+
+const getPoolMarkdown = (task, { formatDate }) => {
+  const { pool, poolMaster = {} } = task.data
+
+  const name = pool.name_label || poolMaster.name_label || UNKNOWN_ITEM
+  const body = [
+    `- **UUID**: ${pool.uuid}`,
+    ...getTemporalDataMarkdown(task.end, task.start, formatDate),
+  ]
+
+  const error = getErrorMarkdown(task)
+  if (error !== undefined) {
+    body.push(error)
+  }
+
+  return {
+    body,
+    title: `[pool] ${name}`,
+  }
 }
 
+const getXoMarkdown = (task, { formatDate, jobName }) => {
+  const body = getTemporalDataMarkdown(task.end, task.start, formatDate)
+
+  const error = getErrorMarkdown(task)
+  if (error !== undefined) {
+    body.push(error)
+  }
+
+  return { body, title: `[XO] ${jobName}` }
+}
+
+const getRemoteMarkdown = async (task, { formatDate, xo }) => {
+  const id = task.data.id
+
+  const name = await xo.getRemote(id).then(
+    ({ name }) => name,
+    error => {
+      logger.warn(error)
+      return UNKNOWN_ITEM
+    }
+  )
+  const body = [
+    `- **ID**: ${id}`,
+    ...getTemporalDataMarkdown(task.end, task.start, formatDate),
+  ]
+
+  const error = getErrorMarkdown(task)
+  if (error !== undefined) {
+    body.push(error)
+  }
+
+  return {
+    body,
+    title: `[remote] ${name}`,
+  }
+}
+
+const getMarkdown = async (task, props) => {
+  const type = task.data?.type
+  if (type === 'pool') {
+    return getPoolMarkdown(task, props)
+  }
+
+  if (type === 'xo') {
+    return getXoMarkdown(task, props)
+  }
+
+  if (type === 'remote') {
+    return getRemoteMarkdown(task, props)
+  }
+}
+
+const toMarkdown = parts => {
+  const lines = []
+  let indentLevel = 0
+
+  const helper = part => {
+    if (typeof part === 'string') {
+      lines.push(`${INDENT.repeat(indentLevel)}${part}`)
+    } else if (Array.isArray(part)) {
+      ++indentLevel
+      part.forEach(helper)
+      --indentLevel
+    }
+  }
+  helper(parts)
+
+  return lines.join('\n')
+}
```

Comment:
So, what do you think about this?

Reply:
Powerful :+1: i have other suggestion no to the function but to the UI to keep it simple.

Reply:
Use markdown titles instead of add indents.

Reply:
We'll see in another PR.

This one implements the support for metadata backups, you can do another one to improve the formatting.